build/mvnd

```diff
@@ -25,7 +25,7 @@ _CALLING_DIR="$(pwd)"
 _COMPILE_JVM_OPTS="-Xms2g -Xmx2g -XX:ReservedCodeCacheSize=1g -Xss128m"
 
 if [ "$CI" ]; then
-  export MAVEN_CLI_OPTS="-Dmvnd.minThreads=8 --no-transfer-progress --errors --fail-fast -Dstyle.color=always"
+  export MAVEN_CLI_OPTS="-Dmvnd.minThreads=4 --no-transfer-progress --errors --fail-fast -Dstyle.color=always"
```

Comment:
can we just use nproc to get the num of available processors

Reply:
`--threads` supports getting by the num of available processors, but `mvnd.minThreads` does not. The config `mvnd.minThreads` is used on purpose as the fallback, and if `--threads` set later in command, the `mvnd.minThreads` will be ignored.

Reply:
I mean

export MAVEN_CLI_OPTS="-Dmvnd.minThreads=$(nproc) --no-transfer-progress --errors --fail-fast -Dstyle.color=always"

Reply:
Good suggestion. Will do it in a follow-up if necessary in future.
This PR mainly concerns lower the memory pressure on build concurrency. Let's see whether it helps this time.